WTG.Analyzers/Analyzers/Emit/EmitConversionVisitor.cs

```diff
@@ -27,16 +27,22 @@ public override ExpressionSyntax VisitLiteralExpression(LiteralExpressionSyntax
 		{
 			if (node.Kind() == SyntaxKind.NumericLiteralExpression)
 			{
-				switch (operandType)
+				try
+				{
+					switch (operandType)
+					{
+						case OpCodeOperand.InlineI:
+							return ExpressionSyntaxFactory.CreateLiteral(Convert.ToInt32(node.Token.Value, CultureInfo.InvariantCulture));
+						case OpCodeOperand.InlineI8:
+							return ExpressionSyntaxFactory.CreateLiteral(Convert.ToInt64(node.Token.Value, CultureInfo.InvariantCulture));
+						case OpCodeOperand.ShortInlineR:
+							return ExpressionSyntaxFactory.CreateLiteral(Convert.ToSingle(node.Token.Value, CultureInfo.InvariantCulture));
+						case OpCodeOperand.InlineR:
+							return ExpressionSyntaxFactory.CreateLiteral(Convert.ToDouble(node.Token.Value, CultureInfo.InvariantCulture));
+					}
+				}
+				catch (OverflowException)
```

Comment:
Is there a nicer way to do this that doesn't involve exception handling?

Reply:
The only way I can think of is to effectively re-implement the conversion logic and add our own bounds checking, I figured it's not worth the added complexity just to avoid the (rare) exception, particularly since it's only an issue during the fix and not during the analysis.